src/engine/Operation.cpp

```diff
@@ -182,8 +183,9 @@ CacheValue Operation::runComputationAndPrepareForCache(
   if (!result.isFullyMaterialized() &&
       !unlikelyToFitInCache(cache.getMaxSizeSingleEntry())) {
     AD_CONTRACT_CHECK(!pinned);
+    auto maxSize = RuntimeParameters().get<"lazy-result-max-cache-size">();
```

Comment:
To be honest I'm not sure if we should take the minimum between this value and `cache.getMaxSizeSingleEntry()` and put this new value into the `unlikelyToFitInCache` function.